Replacing some unicode in R by either different ascii chars etc

JorisGoosen opened the pull request:
- implements https://github.com/jasp-stats/jasp-test-release/issues/984
- these changes should not be copied to the modules' repos because we will just fix `do_gettext` later

JorisGoosen committed:
Try to convert certain paths on windows to short version first (if available)
- Using this for R_HOME was hypothesized to mitigate some problems with spaces or weird characters in username
-- https://github.com/jasp-stats/INTERNAL-jasp/issues/1033
-- https://github.com/jasp-stats/INTERNAL-jasp/issues/778
-- But R seems to get the fullpath anyway later, while it does allow for the library to actually load things like "base"-pkg the engine still crashes afterwards. So marginal improvement at best
-- Im guessing it does not hurt though so Im leaving it in. The shortpath function doesn't do anything if the short path does not exist (and it doesnt need to)
Adding settings for LC_CTYPE="C" on windows
- you can either leave it at default, where it will check if the directory is the same in local codepage as in utf-8, then it sets LC_CTYPE to C.
-- In that case gettext will give nice output
-- Otherwise it will choose local codepage, which makes sure people using a korean system and having korean characters in their username can actually install JASP in their userfolder, gettext stuff will be messed up and certain code might even break. But better than not being able to
start JASP I suppose.
- Or you can specify whether it should be *on* manually
- Added command line arguments for JASP to start in LC_CTYPE C or system mode
-- JASP --setLC_CTYPE_C forces "C" while --setLC_CTYPE_system forces system-default

JorisGoosen committed:
Replacing some unicode in R by either different ascii chars etc
- implements https://github.com/jasp-stats/jasp-test-release/issues/984
- these changes should not be copied to the modules' repos because we will just fix `do_gettext` later

revert H%1$s etc to H0

And now fixing the revert :s

JorisGoosen commented:
After this is merged I'll build a nightly and check it out with some crazy usernames or in some "strange" locale

boutinb commented on JASP-Engine/JASP/R/correlation.R:
```diff
@@ -401,7 +401,7 @@ Correlation <- function(jaspResults, dataset, options){
       
       result$vsmpr <- JASP:::.VovkSellkeMPR(result$p.value)
-      result$vsmpr <- ifelse(result$vsmpr == "∞", Inf, result$vsmpr)
+      result$vsmpr <- ifelse(result$vsmpr == "\u221E", Inf, result$vsmpr)
```
Did you test this??
I have tried, but I cannot get  a situation where this vsmpr  becomes infinite.

JorisGoosen replied:
No I didn't, but I looked up the codepoint for infinity and it is that: https://codepoints.net/U+221E

TimKDJ replied:
This line can just be removed. In https://github.com/TimKDJ/jasp-desktop/commit/9a821494b6deda4800ea8f4538701d88c279007f# I changed the `.VovkSellkeMPR()` function so it returns a numeric value rather than a character.

boutinb commented on JASP-Engine/JASP/po/R-JASP.pot:
```diff
@@ -622,7 +622,7 @@ msgstr ""
 msgstr ""
 
-msgid "Vovk-Sellke Maximum <em>p</em>-Ratio: Based on a two-sided <em>p</em>-value, the maximum possible odds in favor of H%1$s over H%2$s equals 1/(-e <em>p</em> log(<em>p</em>)) for <em>p</em> %3$s .37 (Sellke, Bayarri, & Berger, 2001)."
+msgid "Vovk-Sellke Maximum <em>p</em>-Ratio: Based on a two-sided <em>p</em>-value, the maximum possible odds in favor of H1 over H0 equals 1/(-e <em>p</em> log(<em>p</em>)) for <em>p</em> = .37 (Sellke, Bayarri, & Berger, 2001)."
```
Did you forget to add this change in  MixedModelsCommon.R in your commit? How such a change can come in the pot file, if the original string is not changed?

JorisGoosen replied:
Someone else changed it in `stable` I think, it appeared when I ran the "language generation" thing of jasp.
But maybe I made a mistake?

JorisGoosen replied:
Hmm, I can't see where it came from to be honest. I'll revert that.

TimKDJ commented on JASP-Engine/JASP/R/correlation.R:
```diff
@@ -431,7 +431,7 @@ Correlation <- function(jaspResults, dataset, options){
       }
       result$vsmpr <- JASP:::.VovkSellkeMPR(result$p.value)
-      result$vsmpr <- ifelse(result$vsmpr == "∞", Inf, result$vsmpr)
+      result$vsmpr <- ifelse(result$vsmpr == "\u221E", Inf, result$vsmpr)
```
This line can be removed too

TimKDJ reviewed (changes requested):
So I build your branch, removed all fuzzies and problematic translations, then compiled the translations and tested the two `LC_CTYPE` settings:

```
env.insert("LC_CTYPE",			"C");
```
![withCSpecTranslation](https://user-images.githubusercontent.com/18737241/94255180-230c3a00-ff28-11ea-86d9-024f4ff7a7de.png)

-----
```
//env.insert("LC_CTYPE",			"C");
```
![withoutCSpecTranslation](https://user-images.githubusercontent.com/18737241/94255231-34554680-ff28-11ea-8707-8d1d84d1379a.png)

I guess this isn't going to work properly without our UTF-8 `do_gettext()`.

JorisGoosen commented:
Ah man, not even umlauts works :(
Perhaps the LC_CTYPE being set to C should be an option in the Windows version of JASP?
So that, when it doesn't work they can turn it off?
Bleh

TimKDJ commented:
> Perhaps the LC_CTYPE being set to C should be an option in the Windows version of JASP?
> So that, when it doesn't work they can turn it off?

I guess we could have it as a setting, but I'm doubtful anyone will figure out when they need it unless they post on our GitHub page...

Edit: Although I guess if we make a setting and we do not set it to `C` by default, then for most users this will be an improvement. Only when people use JASP in a different language could it pose a problem, and then they can specify they want it set to `C`. This would only be for one release, right?

JorisGoosen commented:
> Edit: Although I guess if we make a setting and we do not set it to `C` by default, then for most users this will be an improvement. Only when people use JASP in a different language could it pose a problem, and then they can specify they want it set to `C`. This would only be for one release, right?

Ah yes, that could work!
And then when a user switches to german (I'm guessing we didn't put anything strange in dutch) we could:
- Check where JASP is installed, if the path does not contain any non-ascii chars.
- Check whether the username does not contain any non-ascii chars.
If both checks are good we could just turn on LC_CTYPE anyway. 

Maybe we could just do this check anyway and turn it on/off based on that?
It shouldn't be too hard and then we don't need to bother the user with this setting. Although we can add it for manual override under "Advanced - Developer Mode" or something.

And yeah, that would be just for this release.

TimKDJ commented:
Dutch does have stuff like é and ë. There were too many errors in the Dutch translations file so I didn't test that one.
Otherwise I think what you propose sounds like a good solution (with the automatic checks), granted that it actually solves the path problem :P

JorisGoosen commented:
> 
> 
> Dutch does have stuff like é and ë. There were too many errors in the Dutch translations file so I didn't test that one.
> Otherwise I think what you propose sounds like a good solution (with the automatic checks), granted that it actually solves the path problem :P

Hahaha, yeah if it doesn't do that then this whole PR is quite pointless :p

JorisGoosen commented:
So in light of the new setting I'll be adding this PR will be pointless (given that german still looks bad).

JorisGoosen committed:
Add title to windows specific LC_CTYPE box

JorisGoosen commented:
@TimKDJ is right, maybe we want this PR as well.

JorisGoosen commented:
It will have to be rebased after #4311 is merged

JorisGoosen committed:
Requested changes for PR
- getWinLcCtypeSetting moved to Settings with the help of a little forward declaration
- Moved long tooltip to `info` field and typed a bunch of words into the documentation

FBartos commented:
Actually, it seems like that the previous version worked better:
![image](https://user-images.githubusercontent.com/38475991/94921407-8eba4e00-04b8-11eb-9ade-00be25fe754a.png)

than this new one
![image](https://user-images.githubusercontent.com/38475991/94920846-5f571180-04b7-11eb-804a-9d560ec4ae87.png)

However, compiling on my EN PC, the previous version with "š" is also broken

```
    stats$addColumnInfo(name="Variable",   title=gettext("Variable"), type="string")
    stats$addColumnInfo(name="Variable1",  title=gettext("\u03C4"), type="string")
    stats$addColumnInfo(name="Variable2",  title=gettextf("%s a", "\u03C4"), type="string")
    stats$addColumnInfo(name="Variable3",  title=gettextf("%s \u03C4", "a"), type="string")
    stats$addColumnInfo(name="Variable4",  title=gettext("š"), type="string")
    stats$addColumnInfo(name="Variable5",  title=gettextf("%s a", "š"), type="string")
    stats$addColumnInfo(name="Variable6",  title=gettextf("%s š", "a"), type="string")
```

JorisGoosen committed:
Fix crash
- not queued connection but normal, that way the code waits untill engines are restarted
- Also use only 1 signal for changing of LC_CTYPE enum thing in settings

JorisGoosen commented:
Yeah, this one doesn't actually fix anything like, @FBartos this one ought to: https://github.com/jasp-stats/jasp-desktop/pull/4311
But it seems to be behaving weirdly on my system now...

JorisGoosen committed:
- removed qsTr("") from qml because why was it there?
- Make it compile again on a mac
- generated new language files I think?

JorisGoosen committed:
Some more language generating

JorisGoosen committed:
Did I put that extra space there? Well.. It broke the build :(

JorisGoosen committed:
Generating new language files

JorisGoosen commented:
Merged it with the other PR
